enhancements/olm-deletes-operands.md

```diff
@@ -0,0 +1,155 @@
+---
+title: olm-deletes-operands
+authors:
+  - "@hasbro17"
+reviewers:
+  - "@benluddy"
+  - "@ecordell"
+  - "@kevinrizza"
+  - "@njhale"
+approvers:
+  - "@kevinrizza"
+  - "@dmesser"
+  - "@ecordell"
+  - "@shawn-hurley"  
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+see-also:
+  - "N/A"
+replaces:
+  - "N/A"
+superseded-by:
+  - "N/A"
+---
+
+# olm-deletes-operands
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+The purpose of this enhancement is to outline how the [Operator Lifecycle Manager (OLM)][olm] can let operator users opt-in to the clean up of their operands, or operator workloads, upon the uninstallation of their operator. Specifically when uninstalling an operator, OLM would first delete all CustomResources (CRs) managed by an operator to provide the operator time to clean up all associated resources before deleting the operator itself.
+
+## Motivation
+
+By design when OLM uninstalls an operator it does not remove any of the operator’s owned CustomResourceDefinitions (CRDs) and CRs in order to prevent data loss.
+The operator user or cluster admin is expected to manually delete CRDs, CRs and any other secondary resources created by the operator.
+While some secondary resources that have their [OwnerReference][owner-reference] set to a CR would be garbage collected, others such as off-cluster or cross-namespaced resources with no owner references would remain orphaned.
```

Comment:
> While some secondary resources that have their [OwnerReference][owner-reference] set to a CR would be garbage collected, others such as off-cluster or cross-namespaced resources with no owner references would remain orphaned.

This suggests that we propose a way to clean up off-cluster resources.

Reply:
Per the next sentence, I read it as just suggesting that CR deletion should happen while the operator is still running.

Reply:
Yeah this just was meant to point out the expectation from operators to clean up via finalizers, when not everything can be garbage collected. 
But since I haven't explicitly mentioned finalizers until much later I'll reword this to make it clearer.

Reply:
Marking as resolved. Updated the summary to mention that this is solely about deleting CRs while the operator is running to rely on finalizers for the actual cleanup.